dealership/contracts.py

```diff
@@ -1,12 +1,55 @@
 class Contract(object):
-    pass
-
+    def __init__(self, vehicle, customer):
+        self.vehicle = vehicle
+        self.customer = customer
+        self.discount = None
+        
+        if customer.is_employee():
+            self.discount = .90
+        else:
+            self.discount = 1
+            
 
 class BuyContract(Contract):
     def __init__(self, vehicle, customer, monthly_payments):
-        pass
+        super(BuyContract, self).__init__(vehicle, customer)
+        self.monthly_payments = monthly_payments
+            
+    
+    def total_value(self):
+        if self.vehicle.__class__.__name__ == 'Car':
+            return (self.vehicle.sale_price() + (1.07 * self.monthly_payments * \
```

Comment:
Perhaps defining the general formula for calculating the total value first, then writing an if-else statement for the various interest rates might improve readability. This way, the same equation is not used multiple times across the three return statements